File: LojadeCarro/src/main/java/com/javacar/lojadecarro/dto/response/VendaResponse.java
package com.javacar.lojadecarro.dto.response;

import com.javacar.lojadecarro.enums.StatusVenda;
import io.swagger.v3.oas.annotations.media.Schema;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Schema(description = "Resposta para a venda do veículo")
public record VendaResponse(
        @Schema(example = "1", description = "Identificação unica da venda do veiculo")
        Long id,
        @Schema(example = "25000", description = "Valor da venda do veiculo")
        BigDecimal valorVenda,
        @Schema(example = "EM_ANDAMENTO", description = "Status da venda do veiculo")
        StatusVenda statusVenda,
        @Schema(example = "2025-07-01T14:30:00", description = "Data da venda")
        LocalDateTime dataVenda,
        @Schema(example = "{id: 1, marca: Fiat, modelo: Argo}", description = "Dados resumidos do veiculo")
        VeiculoVendaResponse veiculo,
        @Schema(example = "{id: 1, nome: Felipe}", description = "Dados resumidos do vendedor do veiculo")
        UsuarioResumoResponse vendedor,
        @Schema(example = "{id: 1, nome: Goku}", description = "Dados resumidos do comprador do veiculo")
        UsuarioResumoResponse comprador
) {
}

File: LojaDeCarroWeb/src/app/features/veiculos/models/venda-response.ts
import { StatusVeiculo } from  '../../veiculos/models/veiculo-response';

export type StatusVenda =
  | 'EM_ANDAMENTO'
  | 'CONCLUIDA'
  | 'CANCELADA';

export interface VendaRequest {
  veiculoId: number;
}

export interface VeiculoVendaResponse {
  id: number;
  marca: string;
  modelo: string;
  status: StatusVeiculo;
}

export interface UsuarioResumoResponse {
  id: number;
  nome: string;
}

export interface VendaResponse {
  id: number;
  valorVenda: number;
  statusVenda: StatusVenda;
  dataVenda: string;
  veiculo: VeiculoVendaResponse;
  vendedor: UsuarioResumoResponse;
  comprador: UsuarioResumoResponse;
}